src/app/ReadHandler.cpp

```diff
@@ -128,16 +173,35 @@ CHIP_ERROR ReadHandler::OnStatusReport(Messaging::ExchangeContext * apExchangeCo
 
 CHIP_ERROR ReadHandler::SendReportData(System::PacketBufferHandle && aPayload)
 {
+    CHIP_ERROR err = CHIP_NO_ERROR;
     VerifyOrReturnLogError(IsReportable(), CHIP_ERROR_INCORRECT_STATE);
     if (IsInitialReport())
     {
         VerifyOrReturnLogError(mpExchangeCtx != nullptr, CHIP_ERROR_INCORRECT_STATE);
         mSecureHandle = mpExchangeCtx->GetSecureSession();
     }
+    else
+    {
+        mpExchangeCtx = mpExchangeMgr->NewContext(mSecureHandle, this);
+        mpExchangeCtx->SetResponseTimeout(kImMessageTimeoutMsec);
+    }
     VerifyOrReturnLogError(mpExchangeCtx != nullptr, CHIP_ERROR_INCORRECT_STATE);
     MoveToState(HandlerState::Reporting);
-    return mpExchangeCtx->SendMessage(Protocols::InteractionModel::MsgType::ReportData, std::move(aPayload),
-                                      Messaging::SendFlags(Messaging::SendMessageFlags::kExpectResponse));
+    err = mpExchangeCtx->SendMessage(Protocols::InteractionModel::MsgType::ReportData, std::move(aPayload),
```

Comment:
above we move to state Reporting and move back to reportable on failure.

Why has this not been done in previous code?
Would it make sense to instead move to Reporting only on success since the previous state should be 'Reportable' as we do a VerifyOrReturn?

Reply:
just remove that reportable. thanks